fxgl/src/main/kotlin/com/almasb/fxgl/dsl/components/LiftComponent.kt

```diff
@@ -18,104 +18,95 @@ import javafx.util.Duration
  */
 class LiftComponent : Component() {
 
-    /**
-     * Move in X axis.
-     */
-    var isHorizontal = false
+    private class LiftData() {
 
-    /**
-     * Move in Y axis.
-     */
-    var isVertical = false
+        var isHorizontal = false
```

Comment:
Yes, that's what I was thinking of. However, I think `isHorizontal` is not quite correct since we had both horizontal and vertical. So we could move in both horizontal and vertical directions. With the new code, we can only move in one or the other, is that true? Please double check.

We've replaced these with X and Y lift data. So I believe each lift data object should have whether it's on or not:

```
// on Update

if (liftDataX.isOn) {
            if (liftDataX.timer.elapsed(liftDataX.duration)) {
                liftDataX.isGoingPositive = !liftDataX.isGoingPositive
                liftDataX.timer.capture()
            }

            entity.translateX(if (liftDataX.isGoingPositive) liftDataX.speed * tpf else -liftDataX.speed * tpf)
        }

// then do the same with liftDataY ...
```

Even with above it looks like it can be improved since essentially the code is very similar. I wonder if there should be a method inside LiftData that does that ...

Reply:
I compared the current behavior and the one from the state before this PR with the BBoxSample app. And it looks the same.
The vertical movement was now done when `isHorizontal `was set to false. That is why I set it to `false` in the yAxis functions (for liftDataY object). But I agree that this is very confusing. Therefore I would rename this field like you suggested to `isOn` and then each instance has it's on state of it.
I hope I understood you correctly on this.

About having the `LiftData` object solve the code duplication in the update method of LiftComponent we could do something like this:

```java
// inside LiftData class...

        fun translateX(tpf: Double, entity: Entity) {
            if (isOn) {
                negateDirection()
                entity.translateX(if (isGoingPositive) speed * tpf else -speed * tpf)
            }
        }

        fun translateY(tpf: Double, entity: Entity) {
            if (isOn) {
                negateDirection()
                entity.translateY(if (isGoingPositive) speed * tpf else -speed * tpf)
            }
        }

        private fun negateDirection() {
            if (timer.elapsed(duration)) {
                isGoingPositive = !isGoingPositive
                timer.capture()
            }
        }
    }

```

Then we only have to call these functions in the update method like so: 
```java
// inside LiftComponent class...

    override fun onUpdate(tpf: Double) {
        liftDataX.translateX(tpf, entity)
        liftDataY.translateY(tpf, entity)
    }
```
I am not to sure that it is a good idea to let LiftData access the entity object now, though.
I am happy to hear your opinion on this.

Reply:
> I hope I understood you correctly on this.

Yeah, I think so.

> I am not to sure that it is a good idea to let LiftData access the entity object now, though

I agree. How about this:

```
on update() {
    val v = Vec2()
    if (liftDataX.isOn) {
        v.x = liftDataX.update()
    }

    if (liftDataY.isOn) {
        v.y = liftDataY.update()
    }

    entity.transate(v)
}
```

and

```
LiftData::update() {
           if (timer.elapsed(duration)) {
                isGoingPositive = !isGoingPositive
                timer.capture()
            }

     return if (isGoingPositive) speed * tpf else -speed * tpf
}
```

You'd have to double check the logic (sorry, I really should have written a unit test for this), but architecture wise I think we achieved quite a nice result